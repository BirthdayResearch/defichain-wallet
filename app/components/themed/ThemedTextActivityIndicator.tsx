import React from 'react'
import { View } from 'react-native'
import { tailwind } from '@tailwind'
import { ThemedActivityIndicator } from './ThemedActivityIndicator'
import { ThemedText } from './ThemedText'
import { ThemedProps } from '.'

interface ThemedTextActivityIndicatorProps extends ThemedProps {
  message?: string
}

export function ThemedTextActivityIndicator ({ message, light, dark }: ThemedTextActivityIndicatorProps): JSX.Element | null {
  if (message === undefined) {
    return null
  }

  return (
    <View style={tailwind('flex-row justify-center')}>
      <ThemedActivityIndicator />
      <ThemedText
        style={tailwind('ml-2')}
        light={light}
        dark={dark}
      >
        {message}
      </ThemedText>
    </View>
  )
}
