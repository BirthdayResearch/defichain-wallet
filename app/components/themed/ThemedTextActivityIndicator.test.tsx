import { render } from '@testing-library/react-native'
import * as React from 'react'
import { ThemedTextActivityIndicator } from './ThemedTextActivityIndicator'

jest.mock('../../contexts/ThemeProvider')
describe('themed text activity', () => {
  it('should match snapshot', async () => {
    const component = (
      <ThemedTextActivityIndicator
        message='Hello World'
      />
    )
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
