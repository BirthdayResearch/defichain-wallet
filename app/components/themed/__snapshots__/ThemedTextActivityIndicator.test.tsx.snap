// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themed text activity should match snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#ff00af"
    hidesWhenStopped={true}
    size="small"
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "RegularFont",
          "fontSize": 16,
          "fontWeight": "400",
          "lineHeight": 24,
        },
        Array [
          Object {
            "marginLeft": 8,
          },
          Object {
            "color": "rgba(0, 0, 0, 1)",
          },
        ],
      ]
    }
  >
    Hello World
  </Text>
</View>
`;
